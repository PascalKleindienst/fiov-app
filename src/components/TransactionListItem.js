// ==============================================
// TRANSACTION LIST ITEM PAGE COMPONENT =========
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const TransactionListItem = ({ id, description, amount, createdAt }) => (
    <Link className={ amount < 0 ? 'list-item list-item--expense' : 'list-item' } to={`/edit/${id}`}>
        <div>
            <h3 className="list-item__title">{description}</h3>
            <span className="list-item__sub-title">{ moment(createdAt).format('LL') }</span>
        </div>
        <h3 className="list-item__data">
            {numeral(amount / 100).format('-0,0.00$')}
        </h3>
    </Link>
);

// PropTypes
TransactionListItem.propTypes = {
    id: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    createdAt: PropTypes.string.isRequired
};

export default TransactionListItem;