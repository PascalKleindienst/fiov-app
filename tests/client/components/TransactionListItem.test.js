// ==============================================
// TRANSACTIONLISTITEM PAGE TEST ================
// ==============================================
import React from 'react';
import { shallow } from 'enzyme';
import TransactionListItem from '../../../src/components/TransactionListItem';

describe('my transaction list item component', () => {
    test('should render correctly', () => {
        const props = {
            id: 123,
            description: "Some Desc",
            amount: 123456,
            createdAt: "2018-01-01"
        }
        const wrapper = shallow(<TransactionListItem {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render correctly for expenses', () => {
        const props = {
            id: 123,
            description: "Some Desc",
            amount: -123456,
            createdAt: "2018-01-01"
        }
        const wrapper = shallow(<TransactionListItem {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
