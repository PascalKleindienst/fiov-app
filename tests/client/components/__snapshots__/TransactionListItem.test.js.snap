// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my transaction list item component should render correctly 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/123"
>
  <div>
    <h3
      className="list-item__title"
    >
      Some Desc
    </h3>
    <span
      className="list-item__sub-title"
    >
      January 1, 2018
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    1,234.56$
  </h3>
</Link>
`;

exports[`my transaction list item component should render correctly for expenses 1`] = `
<Link
  className="list-item list-item--expense"
  replace={false}
  to="/edit/123"
>
  <div>
    <h3
      className="list-item__title"
    >
      Some Desc
    </h3>
    <span
      className="list-item__sub-title"
    >
      January 1, 2018
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    -1,234.56$
  </h3>
</Link>
`;
